addons/account/views/account_move_views.xml

```diff
@@ -725,18 +725,18 @@
                                                attrs="{
                                                     'required': [('display_type', 'not in', ('line_section', 'line_note'))],
                                                     'invisible': [('display_type', 'in', ('line_section', 'line_note'))],
-                                               }"/>
+                                               }"  optional="show"/>
                                         <field name="partner_id"
                                                domain="['|', ('parent_id', '=', False), ('is_company', '=', True)]"
                                                attrs="{'column_invisible': [('parent.type', '!=', 'entry')]}"/>
                                         <field name="name" widget="section_and_note_text"/>
                                         <field name="analytic_account_id"
                                                groups="analytic.group_analytic_accounting"
-                                               attrs="{'invisible': [('display_type', 'in', ('line_section', 'line_note'))]}"/>
+                                               attrs="{'invisible': [('display_type', 'in', ('line_section', 'line_note'))]}"  optional="show"/>
                                         <field name="analytic_tag_ids"
                                                groups="analytic.group_analytic_tags"
                                                widget="many2many_tags"
-                                               attrs="{'invisible': [('display_type', 'in', ('line_section', 'line_note'))]}"/>
+                                               attrs="{'invisible': [('display_type', 'in', ('line_section', 'line_note'))]}"  optional="show"/>
```

Comment:
we didn't get explicit list of fields to be made optional, but there is a point mentioned in first page of the doc shared in the task regarding optional field's option missing from invoice line's list, so we hid these 3 fields. Anyway, since it is improved meanwhile, I reverted that change now.